app/src/main/java/de/danoeh/antennapod/dialog/VariableSpeedDialog.java

```diff
@@ -66,6 +68,14 @@ public void onStop() {
         EventBus.getDefault().unregister(this);
     }
 
+    @Override
+    public void onDismiss(@NonNull DialogInterface dialog) {
+        super.onDismiss(dialog);
+        if (getActivity().getClass().getName().equals(PlaybackSpeedDialogActivity.class.getName())) {
+            getActivity().finish();
+        }
+    }
+
```

Comment:
This looks a bit hacky to me. Now every time someone writes code for the speed indicator (thinking about the one within the app), they will have to read this and wonder where it comes from. When you instead overwrite the dialog (like I commented above), this would be a lot more "local" change.

Reply:
Agreed, it wasn't looking good, it's completely removed now